src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -673,28 +939,37 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
         }
 
         // This fills in the arguments, refKinds and storesToTemps arrays.
-        private void BuildStoresToTemps(
+        private void VisitAndStoreToTemps(
```

Comment:
>VisitAndStoreToTemps [](http://example.com/codeflow?start=21&length=20)

Conflating visiting and storing to temps feels over complicated, I think these cocerns should be separated.

Reply:
> I think these cocerns should be separated

They were separated before this PR. Unfortunately, we need to do these things at the same time for interpolated strings because we need to have those temps around by the time we get to the interpolated string argument. If you have any suggestions on how to structure it differently I'd be happy to hear them, but I don't really have a good structure in mind.